text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,495 @@
+# OpenTelemetry Telemetry Pipeline Metrics
+
+Propose a uniform standard for telemetry pipeline metrics generated by
+OpenTelemetry SDKs and Collectors.
+
+## Motivation
+
+OpenTelemetry desires to standardize conventions for the metrics emitted by 
+SDKs about success and failure of telemetry reporting. At the same time, the 
+OpenTelemetry Collector has existing conventions which are expected to connect 
+with metrics emitted by SDKs and have similar definitions.
+
+## Explanation
+
+This document proposes two metric instrument semantics and three overall metric 
+instruments for pipeline accounting, considering SDK and Collector pipeline 
+metrics as separate, non-intersecting signals for reasons discussed below. The 
+three instruments are:
+
+- `otelcol_outgoing_items`: Exported, dropped, and discarded items (Collector)
+- `otelcol_incoming_items`: Received and inserted data items (Collector)
+- `otelsdk_outgoing_items`: Exported, dropped, and discarded items (SDK)
+
+The fourth instrument, named `otelsdk_incoming_items`, is logically 
+well-defined by this proposal, but not necessary or useful, for SDK-specific 
+reasons discussed below.
+
+We use the term "pipeline" to describe a directional arrangement of system 
+components ("services") which telemetry flows through. Telemetry is incoming to 
+a service either by receiving or inserting it. It is outgoing when it is sent 
+to a downstream service, dropped, or discarded.
+
+Categories of services defined in this specification are:
+
+- OpenTelemetry SDKs: As telemetry producers, these components are the
+  start of a pipeline.
+- OpenTelemetry Collectors: The OpenTelemetry Collector contains an
+  arrangement of Collector components which receive, process, and export 
+  telemetry.
+
+The terms "following"/"follower" and "preceding"/"preceder" refer to the 
+relative services after or before another service in a pipeline. The preceding 
+service ("preceder") transmits telemetry to the following service ("follower").
+
+An arrangement of discrete pieces of functionality ("components") within a 
+service acting as a single unit is called a segment. Segments can be found in 
+the OpenTelemetry Collector as Collector pipelines. A Collector can contain 
+multiple Collector pipelines which can contain multiple segments. Each segment 
```

Comment:
The segments would be defined as:
- `r1, r2, p1, p2, e1`
- `r1, r2, p1, p2, e2`

Reply:
If I understand correctly, each collector pipeline contains a segment per exporter, where each of these segments contains all the receivers and all the processors of the pipeline, in addition to the exporter.

Can we update the language in this section to state this more clearly? Currently, it reads as "a receiver, zero or more processors, and an exporter" which doesn't appear accurate.

Can we also state "Components can be a part of multiple segments" before describing the relationship between segments and collector pipelines, since it's a prerequisite to understanding?